MoreLinq/AggregateRight.cs

```diff
@@ -0,0 +1,117 @@
+        /// The specified seed value is used as the initial accumulator value.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <typeparam name="TAccumulate">The type of the accumulator value.</typeparam>
+        /// <param name="source">Source sequence.</param>
+        /// <param name="seed">The initial accumulator value.</param>
+        /// <param name="func">An right-associative accumulator function to be invoked on each element.</param>
+        /// <returns>The final accumulator value.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = Enumerable.Range(1, 5);
+        /// string result = numbers.AggregateRight("6", (a, b) => string.Format("({0}/{1})", a, b));
+        /// </code>
+        /// The <c>result</c> variable will contain <c>"(1/(2/(3/(4/(5/6)))))"</c>.
+        /// </example>
+        public static TAccumulate AggregateRight<TSource, TAccumulate>(this IEnumerable<TSource> source, TAccumulate seed, Func<TSource, TAccumulate, TAccumulate> func)
+        {
+            if (source == null) throw new ArgumentNullException("source");
+            if (func == null) throw new ArgumentNullException("func");
+
+            IList<TSource> e = (source as IList<TSource>) ?? source.ToArray();
+
+            return AggregateRightImp(e, seed, func, e.Count);
+        }
+
+        /// <summary>
+        /// Applies an right-associative accumulator function over a sequence.
+        /// The specified seed value is used as the initial accumulator value, 
+        /// and the specified function is used to select the result value.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <typeparam name="TAccumulate">The type of the accumulator value.</typeparam>
+        /// <typeparam name="TResult">The type of the resulting value.</typeparam>
+        /// <param name="source">Source sequence.</param>
+        /// <param name="seed">The initial accumulator value.</param>
+        /// <param name="func">An right-associative accumulator function to be invoked on each element.</param>
+        /// <param name="resultSelector">A function to transform the final accumulator value into the result value.</param>
+        /// <returns>The transformed final accumulator value.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = Enumerable.Range(1, 5);
+        /// int result = numbers.AggregateRight("6", (a, b) => string.Format("({0}/{1})", a, b), str => str.Length);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>21</c>.
+        /// </example>
+        public static TResult AggregateRight<TSource, TAccumulate, TResult>(this IEnumerable<TSource> source, TAccumulate seed, Func<TSource, TAccumulate, TAccumulate> func, Func<TAccumulate, TResult> resultSelector)
```

Comment:
Since the implementation of this is trivial, I thought would be cool provide symmetry with `Aggregate`. If someone expect for use this overload, he will not be disappointed :)